Sources/ClientRuntime/Networking/Http/Middlewares/FlexibleChecksumsRequestMiddleware.swift

```diff
@@ -0,0 +1,86 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0.
+
+public struct FlexibleChecksumsRequestMiddleware<OperationStackInput, OperationStackOutput>: Middleware {
+
+    public let id: String = "FlexibleChecksumsRequestMiddleware"
+
+    let checksumAlgorithms: [String]
+
+    public init(checksumAlgorithms: [String]) {
+        self.checksumAlgorithms = checksumAlgorithms
+    }
+
+    public func handle<H>(context: Context,
+                          input: SerializeStepInput<OperationStackInput>,
+                          next: H) async throws -> OperationOutput<OperationStackOutput>
+    where H: Handler,
+    Self.MInput == H.Input,
+    Self.MOutput == H.Output,
+    Self.Context == H.Context {
+        
+        // Initialize logger
+        guard let logger = context.getLogger() else {
+            return try await next.handle(context: context, input: input)
+        }
+
+        // Get supported list of checksums in priority order
+        let validationList = HashFunction.fromList(checksumAlgorithms).getPriorityOrderValidationList()
+        
+        // Skip flexible checksums workflow if no valid checksum algorithms are provided
+        guard let checksumAlgorithm: HashFunction = validationList.first else {
+            logger.info("Found no supported checksums! Skipping flexible checksums workflow...")
```

Comment:
Not sure, should this be logged as an error?

Reply:
Yeah let's log this as an error

Reply:
Was this addressed?  Still looks like `logger.info()` to me

Reply:
Oops I think I might've reverted this change back accidentally, will change this

Reply:
@jbelkins Thinking about it some more, I believe this should be a logger.info as it is not an unexpected case. The FlexibleChecksumsRequestMiddleware should be added to any method that has the httpChecksum trait with a requestAlgorithmMember enum. But the user does not have to use checksums, it is an opt-in feature.